### PR TITLE
#860 -  Choose Activities and Hobbies

### PR DESCRIPTION
**Pull request by RaccoonBoryvitter**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/EventsExpress/EventsExpress/projects/2)

## Code reviewers

- [ ] @iuriikhrystiuk 
- [ ] @sand0 

### Second Level Review

- [ ] @Apostall107 

## Summary of issue

During the registration process a new user can choose any hobbies so that he or she will be able to join events they're interested in. Each group (or so called activities) contains a list of hobbies. User taps on any group and choose any hobbies from the appeared list. Then user can select other activities or uncheck any selected hobbies. This step is not required during the registration process and then user can select these activities later in Profile Settings.

## Summary of change

- added new entities - CategoryGroups (Activities) for grouping existing Categories (Hobbies);
- provided all required back-end mechanisms to access the data from the database;
- added front-end part - a whole step on the registration process with markable hobbies;
- provided storing selected hobbies into Redux Form.

## Testing approach

ToDo

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

#859 
#860

### CODE REVIEW

File: EventsExpress.Core/IServices/ICategoryGroupService.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace EventsExpress.Core.IServices
+{
+    public interface ICategoryGroupService
+    {
+        IEnumerable<CategoryGroupDto> GetAllGroups();
+
+        CategoryGroupDto GetById(Guid id);
+
+        bool Exists(Guid id);
```
Comment (iuriikhrystiuk) [inline EventsExpress.Core/IServices/ICategoryGroupService.cs:13]: Why do you need these exists methods?

File: EventsExpress.Core/Services/CategoryGroupService.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                                .OrderBy(cg => cg.Title);
+
+            return groups;
+        }
+
+        public CategoryGroupDto GetById(Guid id)
+        {
+            var res = Context.CategoryGroups.Find(id);
+
+            return new CategoryGroupDto { Id = res.Id, Title = res.Title };
```
Comment (iuriikhrystiuk) [inline EventsExpress.Core/Services/CategoryGroupService.cs]: You should use automapper if you are doing mapping multiple times.

File: EventsExpress/EventsExpress.csproj [hunk 3]
```diff
@@ -64,6 +64,10 @@
     <ProjectReference Include="..\EventsExpress.Core\EventsExpress.Core.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (iuriikhrystiuk) [inline EventsExpress/EventsExpress.csproj]: Remove this itemgroup please.

File: EventsExpress.Core/IServices/ICategoryGroupService.cs (hunk 1 above)
Comment (RaccoonBoryvitter) [reply EventsExpress.Core/IServices/ICategoryGroupService.cs:13]: It's rather considered for the future when we can get the group by id, but Exists method is really useless, so yea, I'm on my way

File: EventsExpress.Core/DTOs/CategoryDTO.cs [hunk 4]
```diff
@@ -8,6 +8,9 @@ public class CategoryDto
 
         public string Name { get; set; }
 
+        // public Guid CategoryGroupId { get; set; }
```
Comment (iuriikhrystiuk) [inline EventsExpress.Core/DTOs/CategoryDTO.cs]: Remove commented lines.

File: EventsExpress.Core/Services/CategoryService.cs [hunk 5]
```diff
@@ -24,26 +24,31 @@ public Category GetById(Guid id)
...
         }
 
-        public IEnumerable<CategoryDto> GetAllCategories()
+        public IEnumerable<CategoryDto> GetAllCategories(Guid? groupId)
         {
             var categories = Context.Categories
                                 .Include(c => c.Users)
                                 .Include(c => c.Events)
+                                .Where(c => !groupId.HasValue || c.CategoryGroupId == groupId)
                                 .Select(x => new CategoryDto
```
Comment (iuriikhrystiuk) [inline EventsExpress.Core/Services/CategoryService.cs]: You should use automapper here as well.

**Review by iuriikhrystiuk — APPROVED** [review]

**Pull request merged**